rp2040-hal/src/pio.rs

```diff
@@ -100,7 +100,19 @@ impl<P: Instance> PIO<P> {
     fn add_program(&self, instructions: &[u16], origin: Option<u8>) -> Option<usize> {
         if let Some(offset) = self.find_offset_for_instructions(instructions, origin) {
             for (i, instr) in instructions.iter().enumerate() {
-                self.pio.instr_mem[i + offset].write(|w| unsafe { w.bits(*instr as u32) })
+                let instr = if instr & 0xe000 != 0 {
```

Comment:
it would be cool if we could use the pio crate to encode/decode this so we aren't doing manual bit twiddling, but that can be a TODO.

Reply:
I can add a TODO when I'll update this PR to match the latest changes at [pio-rs#3](https://github.com/rp-rs/pio-rs/pull/3).